### PR TITLE
feat: allow file streaming

### PR DESCRIPTION
**Pull request by roziscoding**
I couldn't find an actually good name for the method, so we can discuss it here, or just move on with this one, which I think is descriptive enough.

### CODE REVIEW

File: src/deps.deno.ts [hunk 1]
```diff
@@ -8,12 +8,24 @@ export type { File } from "https://lib.deno.dev/x/grammy@v1/types.ts";
...
+    return body;
+}
 // Create a temporary file, and copy a local file to a file path
 export const { makeTempFile: createTempFile, copyFile } = Deno;
+// Streams the repsonse body of a URL
+export async function* streamFile(url: string) {
+    const body = await fetchResponseBody(url);
+    for await (const chunk of body) {
+        yield chunk;
+    }
```
Comment (KnorpelSenf) [inline src/deps.deno.ts:24]:
```suggestion
    yield* body;
```

File: src/deps.node.ts [hunk 2]
```diff
@@ -1,7 +1,8 @@
-import * as path from "path";
 import * as fs from "fs";
-import * as os from "os";
+import { IncomingMessage } from "http";
```
Comment (KnorpelSenf) [inline src/deps.node.ts:2]:
```suggestion
import type { IncomingMessage } from "http";
```

File: src/files.ts [hunk 3]
```diff
@@ -43,6 +44,30 @@ export interface FileX {
...
+     *      // Print the size of each chunk
+     *      for await (const chunk of data) {
+     *        console.log(`Read ${chunk.length} bytes`);
+     *      }
+     *  });
+     * ```
+     *
+     * @returns An iterator for the received data
+     */
+    getIterator(): AsyncGenerator<Uint8Array>;
```
Comment (KnorpelSenf) [inline src/files.ts:70]:
```suggestion
    stream(): AsyncGenerator<Uint8Array>;
```
is the better name so let's align the code with the docs

File: src/files.ts [hunk 4]
```diff
@@ -68,6 +93,10 @@ export function installFileMethods(
             else await downloadFile(url, path);
             return path;
         },
+        getIterator() {
```
Comment (KnorpelSenf) [inline src/files.ts:96]:
```suggestion
        stream() {
```
needs to be adjusted, as well

**Review by KnorpelSenf — CHANGES_REQUESTED** [review]

File: src/deps.deno.ts [hunk 5]
```diff
@@ -8,12 +8,24 @@ export type { File } from "https://lib.deno.dev/x/grammy@v1/types.ts";
 
 // Determine whether a file path is absolute
 export { isAbsolute as isAbsolutePath } from "https://deno.land/std@0.135.0/path/mod.ts";
+// Perform a request and returns the raw body
+async function fetchResponseBody(url: string) {
+    const { body } = await fetch(url);
+    if (body === null) throw new Error("Download failed, no response body!");
```
Comment (wojpawlik) [inline src/deps.deno.ts:14]: v2 could reuse [`InputError`](https://github.com/grammyjs/grammY/blob/32b88e25027c33d7778c0a5c20ecc49f963e8a4c/src/core/input_file.ts#L124) here, should I rename it to `ResponseError`?

File: src/deps.deno.ts (hunk 1 above)
Comment (wojpawlik) [reply src/deps.deno.ts:24]: The whole function is unnecessary. `ReadableStream` is already `AsyncIterable`, the new public method just needs to `yield* await`.

File: src/files.ts (hunk 3 above)
Comment (wojpawlik) [reply src/files.ts:70]:
```suggestion
    [Symbol.asyncIterator](): AsyncIterator<Uint8Array>;
```

**Review by wojpawlik — CHANGES_REQUESTED** [review]

File: src/files.ts (hunk 3 above)
Comment (wojpawlik) [reply src/files.ts:70]: Or
```suggestion
    stream(): ReadableStream<Uint8Array>;
```

File: src/deps.deno.ts (hunk 5 above)
Comment (KnorpelSenf) [reply src/deps.deno.ts:14]: Sure

File: src/deps.deno.ts [hunk 6]
```diff
@@ -9,23 +9,20 @@ export type { File } from "https://lib.deno.dev/x/grammy@v1/types.ts";
...
 export const { makeTempFile: createTempFile, copyFile } = Deno;
-// Streams the repsonse body of a URL
-export async function* streamFile(url: string) {
-    const body = await fetchResponseBody(url);
-    for await (const chunk of body) {
-        yield chunk;
-    }
+export async function* readFile(path: string): AsyncIterable<Uint8Array> {
+    const file = await Deno.open(path);
+    yield* file.readable;
```
Comment (KnorpelSenf) [inline src/deps.deno.ts:21]:
```suggestion
export async function readFile(path: string): AsyncIterable<Uint8Array> {
    const file = await Deno.open(path);
    return file.readable;
```

Comment (wojpawlik) [reply src/deps.deno.ts:21]: Will do that in v2; this keeps the return type consistent among `deps.*.ts`.

Comment (KnorpelSenf) [reply src/deps.deno.ts:21]: Now that I think about it, I don't think we have to release this for 1.x. We can change the target branch to a new `v2` branch and ship these changes alongside the necessary adjustment that we will perform for grammY 2.0.

Comment (wojpawlik) [reply src/deps.deno.ts:21]: I have bigger refactoring plans :) Please release a new minor version, I don't think I'm permitted to.

Comment (KnorpelSenf) [reply src/deps.deno.ts:21]: You still could've applied these changes. They don't affect the return type.

Comment (wojpawlik) [reply src/deps.deno.ts:21]: They do! Return type of Deno's `readFile` would become `Promise<AsyncIterable<Uint8Array>>`.

Comment (KnorpelSenf) [reply src/deps.deno.ts:21]: Meh

**Pull request merged**